src/pymor/models/interface.py

```diff
@@ -47,12 +48,95 @@ def __init__(self, products=None, error_estimator=None, visualizer=None,
 
         self.__auto_init(locals())
 
-    @abstractmethod
-    def _solve(self, mu=None, return_output=False, **kwargs):
-        """Perform the actual solving."""
-        pass
+    def _compute(self, solution=False, output=False,
+                 solution_error_estimate=False, output_error_estimate=False,
+                 mu=None, **kwargs):
+        return {}
 
-    def solve(self, mu=None, return_output=False, **kwargs):
+    def _compute_solution(self, mu=None, **kwargs):
+        raise NotImplementedError
+
+    def _compute_output(self, solution, mu=None, **kwargs):
+        if not hasattr(self, 'output_functional'):
+            raise NotImplementedError
+        if self.output_functional is None:
+            raise ValueError('Model has no output')
+        return self.output_functional.apply(solution, mu=mu)
+
+    def _compute_solution_error_estimate(self, solution, mu=None, **kwargs):
+        if self.error_estimator is None:
+            raise ValueError('Model has no error estimator')
+        return self.error_estimator.estimate_error(solution, mu, self)
+
+    def _compute_output_error_estimate(self, solution, mu=None, **kwargs):
+        if self.error_estimator is None:
+            raise ValueError('Model has no error estimator')
+        return self.error_estimator.estimate_output_error(solution, mu, self)
+
+    _compute_allowed_kwargs = frozenset()
+
+    def compute(self, solution=False, output=False,
+                solution_error_estimate=False, output_error_estimate=False, *,
```

Comment:
I've never seen this syntax, can you quickly explain what the `*` argument does?

Reply:
https://www.python.org/dev/peps/pep-3102/

Reply:
Ah, thx. May I understand this simply as "this denotes the end of the list of positional parameters"?

Reply:
@ftalbrecht, it forces you to pass `mu` as a keyword argument ..